byway/src/main/webapp/routepage.js

```diff
@@ -69,60 +100,115 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
       window.alert("Could not calculate route due to: " + status);
     }
   });
+  // add starting point back into schedule panel for better visuals
+
+  route.splice(0,0,start);
+  console.log(route);
+  renderRouteList();
 }
 
-/** Clear the stops panel in the html */
-function clearStops(){
-  const stopList = document.getElementById('stop-list');
-  if(stopList != null){
-    stopList.innerText = "";
+/**
+ * Calculates and sums up the distance and time duration between all destinations (legs)
+ * @param {response} result response from the directions service object
+ */
+function computeDistanceTime(result) {
+  let totalDist = 0;
+  let totalTime = 0;
+  // full route
+  let route_response = result.routes[0];
+  let waypoint_order = route_response.waypoint_order;
+  
+  let route_copy = [...route];
+  for(let i = 0; i < route.length; i++){
+    route[i] = route_copy[waypoint_order[i]];
```

Comment:
What is this doing exactly? Is it re-ordering the route list based on the waypoint order?

Additionally, is the `route` array always guaranteed to be the same size as the `waypoint_order`? Doesn't this function get called when a user removes a waypoint? Please add some additional documentation for this section.

Reply:
I moved it out to another function as per your comment above and added documentation. Sorry about that, I forgot to comment that section altogether.